### PR TITLE
Feature: Add Postgres SSLMODE connection option

### LINKED ISSUE #282: [Feature] Allow Postgres SSL Connections
## Expected behavior
Would like to be able to use `sslmode=require` on Postgres connections to connect to cloud database providers.

However, I'm unable to get it to work with the current connection settings. I'll be happy to submit a pull request, I just haven't been able to get anything to work yet by modifying `/src/Plugin/Pgsql.php` yet.

**Clarification**
I am looking how to connect the main php-censor database to Postgres via SSL, not necessarily for running individual tests. Although that might not be a bad part B to this. (I was looking int he Pgsql plugin, thinking that perhaps php-censor used it to make connections as well.)

I have started work on this in PR #283

### PR DESCRIPTION
**Pull request by mikebronner**
## Description of change

The goal of this PR is to allow for a Postgres connection over SSL for the main php-censor database.

### Next Steps
- [X] Add steps to install command.
- [X] Test install command.
- [x] Add logic for `sslmode` to regular database connection.
- [x] Test migration of install command.
- [x] Test functionality of site once installed.
- [x] Test non-interactive install command.

### Challenges
- [x] Figure out how to get Phynx to run migrations with `sslmode=require` in its connection string.

Fixes #282

### CODE REVIEW

**Comment by corpsee** [comment]
Issue #282. @mikebronner It isn't clear... Do you want to use SSL connection with main PHP Censor DB or with Postgres plugin? It isn't same.

**Comment by mikebronner** [comment]
@corpsee Sorry, and I was looking in the wrong place. I am trying to use it with the main php-censor database.

File: src/Command/InstallCommand.php [hunk 1]
```diff
@@ -451,11 +466,17 @@ protected function getDatabaseInformation(InputInterface $input, OutputInterface
...
         if (isset($db['servers']['write'][0]['port'])) {
             $dns .= ';port=' . (integer)$db['servers']['write'][0]['port'];
         }
+
         $dns .= ';dbname=' . $db['name'];
 
+        if ($db["type"] === "pgsql") {
+            $dns .= ';sslmode=' . $db['servers']['write'][0]['sslmode'];
+        }
+
```
Comment (mikebronner) [inline src/Command/InstallCommand.php]: When trying to connect via SSL, getting the following error:
```
PHP Censor could not connect to database with the details provided. Please try again.
SQLSTATE[08006] [7] sslmode value "require" invalid when SSL support is not compiled in
```
Code looks correct, will follow up on resolving this issue.

Comment (mikebronner) [reply src/Command/InstallCommand.php]: This error is Apple's libraries not having SSL support packed in. The connection test succeeded when running in a different environment (Homestead).

**Comment by mikebronner** [comment]
One problem is in the dependency used for the migration database connection: I'm trying to figure out how to have the Phynx package build the connection to run the migrations with `sslmode=require`.

**Comment by mikebronner** [comment]
@corpsee Would you mind taking a look at this PR? I have it in a functional state at this point. Any feedback welcome. Thanks! :)

File: src/Command/InstallCommand.php [hunk 2]
```diff
@@ -45,6 +45,7 @@ protected function configure()
             ->addOption('db-name', null, InputOption::VALUE_OPTIONAL, 'Database name')
             ->addOption('db-user', null, InputOption::VALUE_OPTIONAL, 'Database user')
             ->addOption('db-password', null, InputOption::VALUE_OPTIONAL, 'Database password')
+            ->addOption('db-sslmode', null, InputOption::VALUE_OPTIONAL, 'Postgres SSLMODE option')
```
Comment (corpsee) [inline src/Command/InstallCommand.php]: Could you rename option to `db-pgsql-sslmode` please?

**Comment by corpsee** [comment]
Also could you add infotmation about new option to application config [documentation](https://github.com/php-censor/php-censor/blob/master/docs/en/configuring_application.md)?

File: src/Command/InstallCommand.php [hunk 3]
```diff
@@ -424,7 +432,14 @@ protected function getDatabaseInformation(InputInterface $input, OutputInterface
             ]
         ];
 
+        if (strtolower($dbType) === "pgsql"
+            && $dbSslmode
+        ) {
+            $dbServers[0]['sslmode'] = $dbSslmode;
```
Comment (corpsee) [inline src/Command/InstallCommand.php]: Could you rename option to `pgsql-sslmode` please?

File: src/Command/InstallCommand.php [hunk 4]
```diff
@@ -401,6 +402,13 @@ protected function getDatabaseInformation(InputInterface $input, OutputInterface
             $dbPort       = $helper->ask($input, $output, $questionPort);
         }
 
+        if (strtolower($dbType) === "pgsql"
+            && ! $dbSslmode = $input->getOption('db-sslmode')
+        ) {
+            $questionSslmode = new Question('Please enter the connection\'s SSL mode (default: prefer): ', 'prefer');
```
Comment (corpsee) [inline src/Command/InstallCommand.php]: I want to release your PR with version `1.1.0` and this question backwards incompatible with old installer. I think we should replace the code block with something without interactive question for this option like:
```php
if (strtolower($dbType) === "pgsql" && ! $dbSslmode = $input->getOption('db-sslmode') {
    $dbSslmode = 'prefer';
}
```
If you want to set this option you can do it with `--db-pgsql-sslmode`.

**Comment by corpsee** [comment]
@mikebronner Thank you for the great work! :+1:

**Comment by mikebronner** [comment]
@corpsee Thanks for the feedback. I believe I have made all requested changes. Let me know if I missed anything. :)

**Comment by corpsee** [comment]
@mikebronner Good. Thanks!

**Comment by corpsee** [comment]
PR merged to `master` branch and will be released with version `1.1.0`.

**Pull request merged**